src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/SetAbstractDomain.cs

```diff
@@ -54,11 +54,11 @@ private static ImmutableHashSet<T> MergeOrIntersect(ImmutableHashSet<T> value1,
         {
             if (value1.IsEmpty)
             {
-                return value2;
+                return merge ? value2 : value1;
```

Comment:
@dotpaul Wonder how this bug was not caught before :) I have a unit test for this now.

Reply:
Also fix in 2.9.x?

Reply:
Yes, sure. Would you prefer I retarget the PR to 2.9.x for other changes or are fine with me just cherry-picking this change?

Reply:
I suppose retarget to 2.9.x so that DFA stays as similar as possible between branches.

Reply:
Sure, let me give that a shot.

Reply:
@dotpaul I tried, but there are quite a few API changes from 2.9.x to master, and lot of nullability errors too. I'll instead create a separate PR to port the changes to base DFA types to 2.9.x